src/topology/fiber_bundle.lean

```diff
@@ -361,6 +361,19 @@ lemma map_proj_nhds (ex : x ∈ e.source) : map proj (𝓝 x) = 𝓝 (proj x) :=
 by rw [← e.coe_fst ex, ← map_congr (e.coe_fst_eventually_eq_proj ex), ← map_map, ← e.coe_coe,
   e.to_local_homeomorph.map_nhds_eq ex, map_fst_nhds]
 
```

Comment:
I quite like the idea of building this by filling in some missing API for `local_homeomorph`. For example one could start by defining:
```lean
def local_homeomorph.homeomorph_of_image_subset_source
  {α β : Type*} [topological_space α] [topological_space β] (e : local_homeomorph α β)
  {s : set α} {t : set β} (hs : s ⊆ e.source) (ht : e '' s = t) :
  s ≃ₜ t :=
sorry
```
and then proceed as:
```lean
def preimage_singleton_homeomorph {b : B} (hb : b ∈ e.base_set) : proj ⁻¹' {b} ≃ₜ F :=
begin
  have h₁ : proj ⁻¹' {b} ⊆ e.source, { sorry, },
  have h₂ : e '' (proj ⁻¹' {b}) = {b} ×ˢ univ, { sorry, },
  refine (e.to_local_homeomorph.homeomorph_of_image_subset_source h₁ h₂).trans _,
  sorry,
end
```
I note we also seem to be missing:
```lean
def source_homeomorph_base_set_prod : e.source ≃ₜ e.base_set × F :=
e.to_local_homeomorph.to_homeomorph_source_target.trans sorry -- Missing `homeomorph` API?
```
What do you think?

Reply:
Seems reasonable.